spec/controller/application_controller_spec.rb

```diff
@@ -78,28 +81,56 @@ def index
   end
 
   describe "#current_form" do
-    let(:form) { build :form, id: 1 }
-
-    before do
-      ActiveResource::HttpMock.respond_to(false) do |mock|
-        mock.get "/api/v1/forms/1", headers, form.to_json, 200
-      end
-    end
-
     it "returns the current form" do
-      params = ActionController::Parameters.new(form_id: 1)
-      allow(Form).to receive(:find).and_return(form)
+      params = ActionController::Parameters.new(form_id: id)
+      allow(Form).to receive(:find).with(id).and_return(form)
       allow(controller).to receive(:params).and_return(params)
       expect(controller.current_form).to eq form
     end
 
     it "memorizes the find form request so it doesn't have to repeat the calls" do
-      params = ActionController::Parameters.new(form_id: 1)
-      allow(Form).to receive(:find).and_return(form)
+      params = ActionController::Parameters.new(form_id: id)
+      allow(Form).to receive(:find).with(id).and_return(form)
       allow(controller).to receive(:params).and_return(params)
       controller.current_form
       controller.current_form
       expect(Form).to have_received(:find).exactly(1).times
     end
   end
+
+  describe "#current_live_form" do
+    it "returns the current live form" do
+      params = ActionController::Parameters.new(form_id: id)
+      allow(Form).to receive(:find_live).with(id).and_return(form)
+      allow(controller).to receive(:params).and_return(params)
+      expect(controller.current_live_form).to eq form
```

Comment:
Very minor and something that was already in this file, but there should be a newline before the expect for readability. That can be added in a separate commit.

Reply:
They've now been separated properly